datahub-web-react/src/app/document/hooks/useDocumentTreeExpansion.ts

```diff
@@ -0,0 +1,155 @@
+import { useCallback, useState } from 'react';
+
+import { DocumentChild, useDocumentChildren } from '@app/document/hooks/useDocumentChildren';
+
+interface DocumentNode {
+    urn: string;
+    title: string;
+    parentUrn?: string | null;
+}
+
+interface UseDocumentTreeExpansionParams {
+    /**
+     * URN of document to exclude from results (e.g., when moving a document)
+     */
+    excludeUrn?: string;
+}
+
+interface UseDocumentTreeExpansionReturn {
+    /** Set of currently expanded document URNs */
+    expandedUrns: Set<string>;
+    /** Map of document URN to whether it has children */
+    hasChildrenMap: Record<string, boolean>;
+    /** Cache of loaded children by parent URN */
+    childrenCache: Record<string, DocumentNode[]>;
+    /** Set of document URNs currently loading children */
+    loadingUrns: Set<string>;
+    /** Function to expand/collapse a document node */
+    handleToggleExpand: (urn: string) => Promise<void>;
+    /** Setter for expandedUrns (for external control) */
+    setExpandedUrns: React.Dispatch<React.SetStateAction<Set<string>>>;
+    /** Setter for hasChildrenMap (for external control) */
+    setHasChildrenMap: React.Dispatch<React.SetStateAction<Record<string, boolean>>>;
+    /** Setter for childrenCache (for external control) */
+    setChildrenCache: React.Dispatch<React.SetStateAction<Record<string, DocumentNode[]>>>;
+    /** Setter for loadingUrns (for external control) */
+    setLoadingUrns: React.Dispatch<React.SetStateAction<Set<string>>>;
+}
+
+/**
+ * Hook to manage document tree expansion state and children loading.
+ * Handles:
+ * - Expanding/collapsing tree nodes
+ * - Loading children on demand
+ * - Caching loaded children
+ * - Tracking which nodes have children
+ * - Loading states
+ *
+ * Used by DocumentTree and MoveDocumentPopover to provide consistent expansion behavior.
+ */
+export function useDocumentTreeExpansion(params: UseDocumentTreeExpansionParams = {}): UseDocumentTreeExpansionReturn {
```

Comment:
hmm i'm not actually seeing this used anywhere

Reply:
Thank you! Removing it..